File: php/src/ChessStrategy.php
<?php
namespace BizX;
final class ChessStrategy {
    public int $turn = 0;
    public array $resources = ['gold'=>100,'food'=>100,'science'=>0];
    public function tick(string $resource, int $amount): int { $this->resources[$resource] = ($this->resources[$resource] ?? 0) + $amount; return ++$this->turn; }
    public function uci(string $fen, string $command='go movetime 100'): array { return ['protocol'=>'UCI','fen'=>$fen,'command'=>$command,'adapter'=>'external-engine']; }
}

File: csharp/ChessStrategy.cs
namespace BizX.Games;

public sealed class ChessStrategy {
    public long Turn { get; private set; }
    public Dictionary<string,long> Resources { get; } = new() {{"gold",100},{"food",100},{"science",0}};
    public long Tick(string resource, long amount) { Resources[resource] = Resources.GetValueOrDefault(resource) + amount; return ++Turn; }
    public object Uci(string fen, string command = "go movetime 100") => new { Protocol="UCI", Fen=fen, Command=command, Adapter="external-engine" };
}
